rpc/src/v1/helpers/signature.rs

```diff
@@ -9,16 +9,16 @@ pub fn verify_signature(is_prefixed: bool, message: Bytes, signature: H520, chai
 	let hash = if is_prefixed {
 		eth_data_hash(message.0)
 	} else {
-		keccak(message.0.clone())
+		keccak(message.0)
 	};
 
 	let signature = Signature::from(signature.0);
 	let is_valid_for_current_chain = chain_id.map(|chain_id| {
 		let v = signature.v() as u64;
 		if v > 36  {
```

Comment:
@seunlanlege 

this if expression is unnecessary now so remove it and return false from the match instead :+1:

Also I would prefer if you can perform ```saturated_mul``` on chain_id * 2 because it can also overflow sorry if I was unclear